### ISSUE TITLE
adding timeout to ssh_knows_hosts.present state. by default it times out after 5 seconds

### ISSUE THREAD

**Issue by aronneagu** [saltstack/salt#29335]
The state ssh_known_hosts.present fails if the host queried doesn't reply in 5 seconds or less. Is it possible to add a timeout option to ssh_known_state? This state actually invokes ssh-keyscan, which supports timeout option (argument -T).

The state i was trying to run is:
```
bitbucket.org:
  ssh_known_hosts:
    - present
    - user: root
    - enc: ssh-rsa
    - fingerprint: 97:8c:1b:f2:6f:14:6b:5c:3b:ec:aa:46:46:74:7c:40
```

The result running the state is:
```
635e0202-484c-4047-96c8-2fbb5d38b887:
----------
          ID: bitbucket.org
    Function: ssh_known_hosts.present
      Result: False
     Comment: Unable to receive remote host key
     Started: 16:30:02.583761
    Duration: 5142.302 ms
     Changes:   

Summary for 635e0202-484c-4047-96c8-2fbb5d38b887
------------
Succeeded: 0
Failed:    1
```

Thank you

**Comment by zerthimon** [comment]
+1

**Comment by zerthimon** [comment]
Would also be great to have a
``` 
 - retry: 3
```
option, together with the timout.
I have git.present states which depend on this heavily.
<Quote>
https://docs.saltstack.com/en/latest/ref/states/all/salt.states.git.html#salt.states.git.present
Important
Before using git over ssh, make sure your remote host fingerprint exists in your ~/.ssh/known_hosts file.
</Quote>

**Comment by LondonAppDev** [comment]
+1 I'm also having this issue (with bitbucket.org specifically).

**Comment by zerthimon** [comment]
@LondonAppDev great to see I'm not alone in my pain with bitbucket.
Their support is selling me BS about latency in my networks :)

**Comment by LondonAppDev** [comment]
@zerthimon ahh I see. Their network does seem a bit slow but at the same time the salt state should support a timeout parameter.

**Comment by zerthimon** [comment]
@LondonAppDev It alredy does (in the develop branch)...
What I did is overloaded the original modules from stable with ssh modules from develop and added the timeout to my states, but even when I set timeout to 1 minute, bitbucket still times out sometimes.

**Comment by LondonAppDev** [comment]
@zerthimon oh sweet. Wow 1 minute is horrendous lag.

**Comment by colegatron** [comment]
There is any official release supporting the timeout? Bitbucket is killing me

**Comment by jfindlay** [comment]
@colegatron, this should be available in 2016.3.0.

**Comment by cachedout** [comment]
Confirmed that timeout option has been added to 2016.3.0. Closing issue.
